### PR TITLE
Access

### PR DESCRIPTION
**Pull request by luandro**
Adds Pirania to Lime-App and stories for all access pages.

### CODE REVIEW

**Review by nicopace — APPROVED** [review]

**Comment by spiccinini** [comment]
Hi! I have to find some time to test it. With @germanferrero we are using selenium now to test (this work is not yet merged).

Why there are thousands of changes to the package-lock file?  (it may be a standard nonsense npm thing).

Anyhow, great to know that this is moving forward :)

**Comment by luandro** [comment]
Awesome work integrating selenium!

Standard npm nonsense is almost right. `package-lock` ensures that all the packages are "locked", so when someone does a `npm i` it almost always changes, as the thousands of npm packages are always changing.

**Comment by spiccinini** [comment]
> Awesome work integrating selenium!
> 
> Standard npm nonsense is almost right. `package-lock` ensures that all the packages are "locked", so when someone does a `npm i` it almost always changes, as the thousands of npm packages are always changing.

After reading a bit we may  be using `npm ci` instead of `npm install` ? I know that in the npm world maybe it is ok to push a new version every 2 seconds. But at the same time this new things may be breaking things, so even if this is using semver they should not be breaking APIs but they may contain new bugs... I always prefer to be carfull when upgrading, specially in projects like this one that have very low automated testing.

**Comment by spiccinini** [comment]
> Awesome work integrating selenium!

If you want to test or see, latest code is now here: https://github.com/germanferrero/lime-packages/commits/integration_tests
We are using the new qemu integration in libremesh for this :)

File: src/components/box/index.js [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
-		<div style={{ background: 'rgb(68, 146, 127)',padding: '10px',color: '#fff' }}>
-			<b>{title}</b>
-		</div>
-		<div style={{ border: '1px solid #ccc',padding: '10px' }}>
-			{children}
-		</div>
-	</div>
-);
+export const Box = ({ title, children, collapse, collapsed }) => {
+	const initialState = !!collapsed;
```
Comment (germanferrero) [inline src/components/box/index.js:6]: Double negation here :)

File: src/config.js [hunk 2]
```diff
@@ -17,6 +19,8 @@ export const plugins = [
 	Notes,
 	Admin,
 	Meta,
-	NetworkAdmin,
-	Fbw,  // fbw does not have menu item
+	// GroundRouting,
```
Comment (germanferrero) [inline src/config.js:22]: We should not removed NetworkAdmin plugin neither add the comments related to GroundRouting back again

File: src/components/select/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { h } from 'preact';
```
Comment (germanferrero) [inline src/components/select/index.js:1]: It looks to me that this select component is too specific to the needs of pirania ui, and so it shouldn't be included as an app-wise component, as it won't be easily reused

File: src/components/input/index.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { h } from 'preact';
```
Comment (germanferrero) [inline src/components/input/index.js:1]: It looks to me that this select component is too specific to the needs of pirania ui, and so it shouldn't be included as an app-wise component, as it won't be easily reused.

**Comment by germanferrero** [comment]
Hi @luandro thanks for all the work on pirania! I would like to keep reviewing this PR.
As a main concern I think we should add the possibility to easily opt-in (or opt-out ?) from including pirania in the limeapp.
Since many community networks will not be using it, and having it included in the LimeApp will be confusing for them.

**Comment by altergui** [comment]
thanks luandro, marcos, nico and everyone else for this! i've tested it and it works nice. some parts look like they are not yet finished (do nothing) so to put this code in production i hid them using "display: none" in 97cb905. accidentally pushed that to this pull request but reverted it afterwards. sorry.

**Comment by germanferrero** [comment]
This PR was deprecated in favor of #329

**Pull request closed without merging**